src/v2/Apps/Partner/PartnerApp.tsx

```diff
@@ -0,0 +1,49 @@
+import React from "react"
+import { Box, Text } from "@artsy/palette"
+import { Footer } from "v2/Components/Footer"
+import { AppContainer } from "../Components/AppContainer"
+import { HorizontalPadding } from "../Components/HorizontalPadding"
+import { NavigationTabsFragmentContainer as NavigationTabs } from "v2/Apps/Partner/Components/NavigationTabs"
+import { createFragmentContainer, graphql } from "react-relay"
+import { PartnerApp_partner } from "v2/__generated__/PartnerApp_partner.graphql"
+import { HttpError } from "found"
+
+export interface PartnerAppProps {
+  partner: PartnerApp_partner
+}
+
+export const PartnerApp: React.FC<PartnerAppProps> = ({
+  partner,
+  children,
+}) => {
+  if (!partner) {
+    throw new HttpError(404)
```

Comment:
We can add a `@principalField` declaration (a la https://github.com/artsy/force/blob/e672299796b540211226b0f1062fcd88ac1058f5/src/v2/Apps/Fair/fairRoutes.tsx#L36) which will automatically handle the 404 behavior

Reply:
That's much better, thanks!